src/unitxt/metrics.py

```diff
@@ -102,6 +101,8 @@ class Metric(Artifact):
     #
     score_prefix: str = ""
 
+    name: Optional[str] = None
```

Comment:
What about using the __id__ field?  Can it work? Prefer not to add another identifier.

Reply:
I tried it, but it's not possible to set it from outside:
```
Value error, Invalid unitxt artifact: Unexpected keyword argument(s) {'__id__': 'test'} for class LLMAsJudge.
Should be one of: ['type', '__description__', '__tags__', '__id__', 'data_classification_policy', ...]
```
It's probably due to the double undescores which are [masked in python](https://docs.python.org/3/tutorial/classes.html#private-variables). I tried prefixing with classname, but that also doesn't work: `"_LLMAsJudge__id__": "test"`

Maybe we can make `__id__` public and overwritable?

Reply:
Hmm, actually it's enough to change it from `InternalField` to `NonPositionalField`, seems it's handled differently in dataclasses. I changed it in `Artifact`

Reply:
I understand. Let's see if  @elronbandel has any suggestion.

Reply:
But I'd like @elronbandel to give his feedback, because it seems (also according to the test), that is was intentional.  Maybe __id__ was only for artifacts in the catalog.

Reply:
This might need to be changed as well:
```python
    @staticmethod
    def artifact_to_jsonable(artifact):
        if artifact.__id__ is None:
            return artifact.to_dict()
        return artifact.__id__
```

https://github.com/IBM/unitxt/blob/31f3abc2581d05221762a60d5e7168d9402fea99/src/unitxt/schema.py#L30